File: app/src/main/java/com/siprikorea/clock/MainActivity.java
package com.siprikorea.clock;

import android.os.Bundle;
import android.os.Handler;
import android.os.Looper;
import android.os.Message;
import android.util.TypedValue;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.Locale;

public class MainActivity extends AppCompatActivity {
    private TextView mTimeView;
    private TextView mDateView;
    private SimpleDateFormat mTimeFormat;
    private SimpleDateFormat mDateFormat;
    static private Handler mTimeHandler;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        mTimeView = findViewById(R.id.time);
        mDateView = findViewById(R.id.date);
        mDateView.setTextSize(TypedValue.COMPLEX_UNIT_SP, mTimeView.getTextSize() / 2);

        mTimeFormat = new SimpleDateFormat("KK:mm:ss", Locale.getDefault());
        mDateFormat = new SimpleDateFormat("yyyy.MM.dd (E)", Locale.getDefault());

        mTimeHandler = new Handler(Looper.getMainLooper()) {
            @Override
            public void handleMessage(@NonNull Message msg) {
                Date date = new Date();
                mTimeView.setText(mTimeFormat.format(date));
                mDateView.setText(mDateFormat.format(date));
            }
        };

        runTime();
    }

    protected void runTime() {
        new Thread(new Runnable() {
            @Override
            public void run() {
                while (true) {
                    try {
                        mTimeHandler.sendMessage(mTimeHandler.obtainMessage());
                        Thread.sleep(1000);
                    } catch (Exception e) {
                        e.printStackTrace();
                    }
                }
            }
        }).start();
    }
}

File: app/src/main/java/com/siprikorea/clock/MainActivity.kt
package com.siprikorea.clock

import android.annotation.SuppressLint
import android.os.Bundle
import android.os.Handler
import android.os.Looper
import android.os.Message
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import java.text.SimpleDateFormat
import java.util.*

class MainActivity : AppCompatActivity() {

    /**
     * on create
     */
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // text view
        val mTimeView: TextView = findViewById(R.id.time)
        val mDateView: TextView = findViewById(R.id.date)

        // time format
        val mTimeFormat = SimpleDateFormat("KK:mm:ss", Locale.ENGLISH)
        val mDateFormat = SimpleDateFormat("yyyy. MM. dd EEE", Locale.ENGLISH)

        // time handler
        mTimeHandler = object : Handler(Looper.getMainLooper()) {
            override fun handleMessage(msg: Message) {
                val date = Date()
                mTimeView.text = mTimeFormat.format(date)
                mDateView.text = mDateFormat.format(date)
            }
        }

        runTimer()
    }

    /**
     * run timer
     */
    private fun runTimer() {
        Thread {
            while (true) {
                try {
                    mTimeHandler!!.sendMessage(mTimeHandler!!.obtainMessage())
                    Thread.sleep(1000)
                } catch (e: Exception) {
                    e.printStackTrace()
                }
            }
        }.start()
    }

    companion object {
        private var mTimeHandler: Handler? = null
    }
}